throttler/throttler.go

```diff
@@ -0,0 +1,78 @@
+package throttler
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+)
+
+//ThrottledAction is a common interface for actions that can be executed
+type ThrottledAction interface {
+	Execute()
+}
```

Comment:
what do you think about removing this abstraction and using `server.Runner` ? cc/ @ronnylt